bots/pr/src/main/java/org/openjdk/skara/bots/pr/PullRequestBot.java

```diff
@@ -224,16 +225,22 @@ private List<PullRequest> getJCheckConfUpdateRelatedPRs() {
                 .filter(key -> !targetRefPRMap.get(key).isEmpty())
                 .toList();
         for (var targetRef : allTargetRefs) {
-            var currConfOpt = remoteRepo.fileContents(".jcheck/conf", targetRef);
-            if (currConfOpt.isEmpty()) {
-                continue;
-            }
-            var currConf = currConfOpt.get();
-            if (!jCheckConfMap.containsKey(targetRef)) {
-                jCheckConfMap.put(targetRef, currConf);
-            } else if (!jCheckConfMap.get(targetRef).equals(currConf)) {
-                ret.addAll(remoteRepo.openPullRequestsWithTargetRef(targetRef));
-                jCheckConfMap.put(targetRef, currConf);
+            try {
+                var currConfOpt = remoteRepo.fileContents(".jcheck/conf", targetRef);
+                if (currConfOpt.isEmpty()) {
+                    continue;
+                }
+                var currConf = currConfOpt.get();
+                if (!jCheckConfMap.containsKey(targetRef)) {
+                    jCheckConfMap.put(targetRef, currConf);
+                } else if (!jCheckConfMap.get(targetRef).equals(currConf)) {
+                    ret.addAll(remoteRepo.openPullRequestsWithTargetRef(targetRef));
+                    jCheckConfMap.put(targetRef, currConf);
+                }
+            } catch (UncheckedRestException e) {
+                if (e.getStatusCode() != 404) {
```

Comment:
```suggestion
                // If the targetRef is invalid, fileContents() will throw a 404 instead of returning
                // empty. In this case we should ignore this and continue processing other PRs.
                // Any invalid refs will get removed from targetRefMap in the next round.
                if (e.getStatusCode() != 404) {
```